docs/FLUTTER_STYLE.md

```diff
@@ -0,0 +1,137 @@
+# Code Style
+
+All rules fall through to the [Effective Dart](https://dart.dev/guides/language/effective-dart)
+guidelines.
+
+## 1. All comments, identifiers, issues, etc. should be written in English
+
+## 2. Limit code to contain 80 characters per line
+
+Exceptions:
+
+- documentation code
+- hard-coded URLs (e.g., in comments)
+- TODO comments should stay on a single line, so that tools can read & report
+  the full message
+
+## 3. Single- vs. Multi-line function calls
+
+Put calls in a single line if they fit:
+
+```dart
+return Container(color: Colors.pink, child: myChild);
+```
+
+Wrap if there are more than two named parameters, they get longer:
+
+```dart
+return Container(
+  color: context.theme.primaryColor,
+  padding: EdgeInsets.symmetric(horizontal: 16),
+  child: myChild,
+);
+```
+
+## 4. Always prefer relative to absolute imports
+
+E.g. `import '../common/module.dart'` instead of `import '/common/module.dart'`
+
+## 5. Use private functions/classes if they are only used locally
+
+In Flutter, you make symbols private by prefixing them with an underscore (e.g. `_DatePickerState`).
+
+## 6. Grouping and ordering concepts within a file
+
+- Group concepts by their functionality vs. e.g. their return type
+- Order these groups of concepts within a file by their relevance
+
+  - E.g. Put helper methods beneath the methods they help
+
+    Why? The name of helper methods should convey their function so you don't
+    need to view their contents to get an initial understanding of the class.
+
+```dart
+import ...
+
+int main() {
+  helper1();
+  helper2();
+}
+
+// Group 1
+int helper1() { return helper11().toInt(); }
+
+String helper11() { return '11' }
+
+// Group 2
+int helper2() { return helper21().toInt(); }
+
+String helper11() { return '21' }
```

Comment:
Is there a reason we put a semicolon after the return in two of these functions but not in the other two?